lib/cart/cart_repository.dart

```diff
@@ -16,14 +17,29 @@ class CartRepository {
   final CartApi _cartApi;
 
   final _cartStreamController = BehaviorSubject<Cart>();
-  final _cartMessageStreamController = BehaviorSubject<CartMessage>.seeded(
-    CartReadyMessage(),
-  );
+  final _cartMessageStreamController =
+      BehaviorSubject<CartMessage>.seeded(CartInitialize());
 
   void init() async {
-    // TODO: proper initialization, find anonymous or user cart, or create one?
-    final cart = await _cartApi.fetchCart(cartCode: kTestCart);
-    _cartStreamController.add(cart);
+    // TODO: fetch previous carts from local storage
+    try {
+      final cart = await _cartApi.fetchCart(cartCode: kTestCart);
+      _cartStreamController.add(cart);
+    } on CartException catch (e) {
+      // if the cart is not found, create a new one
+      if (e.reason == CartExceptionReason.notFound) {
+        _cartMessageStreamController.add(CartNotFound());
+
+        final cart = await _cartApi.createCart();
```

Comment:
What if createCart fails? Can you try to temporarily throw an error in this method and see what happens?

We want that exception to bubble up correctly.

Reply:
Nice catch, the cart creation method should be its own thing